src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -612,10 +602,8 @@ private void RestoreEvents_SolutionRestoreCompleted(SolutionRestoredEventArgs ar
 
         private bool ShouldNoOpDueToRestore(SolutionRestoredEventArgs latestRestore)
         {
-            return latestRestore != null &&
-                   _currentRestore != null &&
-                   latestRestore.SolutionSpecHash == _currentRestore.SolutionSpecHash &&
-                   latestRestore.IsSuccess == _currentRestore.IsSuccess;
+            return _currentRestore != null &&
```

Comment:
You're right this is a mistake (damn those double negatives). It should no-op only when latest restore result is no-op or current result is null.